py/torch_tensorrt/dynamo/conversion/impl/permutation.py

```diff
@@ -34,59 +36,160 @@ def permute(
+        # can be simplified to shifts: [2, 2], dims: [1, 0]
+        for shift, dim in zip(shifts, dims):
+            if dim in shift_dict:
+                shift_dict[dim] += shift
+            else:
+                shift_dict[dim] = shift
+    start = [0] * len(input.shape)
+    for d, s in shift_dict.items():
+        start[d] = get_positive_dim(-s, input.shape[d])
+    return start
+
+
+def calc_start_by_dynamic_shape(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    shifts: Sequence[Union[int, TRTTensor]],
+    dims: Sequence[int],
+) -> TRTTensor:
+    start = [0] * len(input.shape)
+    default_tensor = get_trt_tensor(ctx, 0, name + "_get_0")
+
+    if dims == []:
+        dim_length = impl.shape.shape(ctx, target, source_ir, name + "_shape", input, 1)
+        start[1] = impl.elementwise.sub(
+            ctx, target, source_ir, name + "_sub", dim_length, shifts[0]
+        )
+    else:
+        for d, s in zip(dims, shifts):
+            if isinstance(start[d], TRTTensor):
+                start[d] = impl.elementwise.sub(
+                    ctx, target, source_ir, name + "_sub", start[d], s
+                )
+            else:
+                dim_length = impl.shape.shape(
+                    ctx, target, source_ir, name + "_shape", input, d
+                )
+                start[d] = impl.elementwise.sub(
+                    ctx, target, source_ir, name + "_sub", dim_length, s
+                )
+
+    for idx in range(len(start)):
+        if start[idx] == 0:
+            start[idx] = default_tensor
+    concat_layer = ctx.net.add_concatenation(start)
+    concat_layer.axis = 0
+    set_layer_name(concat_layer, target, f"{name}_gather", source_ir)
+    return concat_layer.get_output(0)
+
+
 def roll(
     ctx: ConversionContext,
     target: Target,
     source_ir: Optional[SourceIR],
     name: str,
     input: TRTTensor,
-    shifts: Union[int, Sequence[int]],
+    shifts: Union[int, Sequence[Union[int, TRTTensor]]],
     dims: Union[int, Sequence[int]],
 ) -> TRTTensor:
-    shape = input.shape
     if isinstance(shifts, int):
         shifts = [shifts]
     if isinstance(dims, int):
         dims = [dims]
 
-    if dims != []:
-        rank = len(shape)
-        start = [0] * rank
-        stride = [1] * rank
-        for i in range(len(dims)):
-            d = dims[i]
-            s = shifts[i]
-            start[d] += get_positive_dim(
-                -s, shape[d]
-            )  # in case that dims has multiple same dim
-
-        layer = ctx.net.add_slice(
+    is_dynamic_shape = has_dynamic_shape(input.shape)
+    if all(isinstance(shift, TRTTensor) for shift in shifts):
+        is_dynamic_shift = False
+    else:
+        is_dynamic_shift = True
+
+    # handle static shape for the input tensor and shifts:
+    if not is_dynamic_shape and not is_dynamic_shift:
+        orignal_shape = input.shape
+        if dims == []:
```

Comment:
if the dims are not provided, it will be default to  []:
https://github.com/pytorch/TensorRT/blob/main/py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py#L3368
```
    return impl.permutation.roll(
        ctx,
        target,
        SourceIR.ATEN,
        name,
        args[0],
        args[1],
        args_bounds_check(args, 2, []),
    )
```